main.nf

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env nextflow
+nextflow.enable.dsl=2
+
+//--------------------------------------------------------------------------
+// Param Checking
+//--------------------------------------------------------------------------
+
+if(!params.fastaSubsetSize) {
+  throw new Exception("Missing params.fastaSubsetSize")
+}
+
+if(params.inputFilePath) {
+  seqs = Channel.fromPath( params.inputFilePath )
+           .splitFasta( by:params.fastaSubsetSize, file:true  )
+}
+else {
+  throw new Exception("Missing params.inputFilePath")
+}
+
+//--------------------------------------------------------------------------
+// Main Workflow
+//--------------------------------------------------------------------------
+
+workflow {
+  tabfiles = findStats(seqs)
+  catFiles(tabfiles.collectFile(), params.outputFileName)
```

Comment:
remove this as "collectFile" does the concatenation.  
collectFile(name: params.outputFileName, storeDir: params.outputDirectory)

Reply:
how does it know to collect tabfiles?

Reply:
it uses "cat" i believe